### ISSUE TITLE
[Bug] 1.19.4 server side crash from generating an "AncientEmeraldIceBlock" due to the particle generation.

### ISSUE THREAD

**Issue by PurpleStar69** [quiqueck/BetterEnd#205]
### What happened?

While testing out the mod (both with and without lithium), I ran into a consistent server side crash relating to the Ancient Emerald Ice Blocks from the Ice Starfields. Apparently the server crashes attempting to generate the particles for these blocks. Text file with the error message attached as well.

This is a 1.19.4 error just fyi, the Github hasn't been updated to list that version of the mod yet.

### BetterEnd

3.2.0

### BCLib

2.3.1

### Fabric API

0.79.0

### Fabric Loader

fabric-server-mc.1.19.4-loader.0.14.19-launcher.0.11.2

### Minecraft

1.19.3

### Relevant log output

```shell
---- Minecraft Crash Report ----
// My bad.

Time: 2023-05-01 13:48:22
Description: Exception ticking world

java.lang.ExceptionInInitializerError
	at org.betterx.betterend.blocks.AncientEmeraldIceBlock.makeParticles(AncientEmeraldIceBlock.java:61)
	at org.betterx.betterend.blocks.AncientEmeraldIceBlock.method_9514(AncientEmeraldIceBlock.java:56)
	at net.minecraft.class_4970$class_4971.method_26199(class_4970.java:993)
	at net.minecraft.class_3218.method_18203(class_3218.java:506)
	at net.minecraft.class_3215.method_14161(class_3215.java:380)
	at net.minecraft.class_3215.method_12127(class_3215.java:324)
	at net.minecraft.class_3218.method_18765(class_3218.java:319)
	at net.minecraft.server.MinecraftServer.method_3813(MinecraftServer.java:901)
	at net.minecraft.class_3176.method_3813(class_3176.java:283)
	at net.minecraft.server.MinecraftServer.method_3748(MinecraftServer.java:828)
	at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:675)
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:266)
	at java.base/java.lang.Thread.run(Thread.java:1623)
Caused by: java.lang.IllegalStateException: Registry is already frozen (trying to add key ResourceKey[minecraft:particle_type / betterend:glowing_sphere])
	at net.minecraft.class_2370.method_40274(class_2370.java:135)
	at net.minecraft.class_2370.method_46744(class_2370.java:141)
	at net.minecraft.class_2370.method_10272(class_2370.java:186)
	at net.minecraft.class_2378.method_39197(class_2378.java:118)
	at net.minecraft.class_2378.method_10230(class_2378.java:114)
	at org.betterx.bclib.particles.BCLParticleType.register(BCLParticleType.java:70)
	at org.betterx.bclib.particles.BCLParticleType.register(BCLParticleType.java:66)
	at org.betterx.betterend.registry.EndParticles.register(EndParticles.java:48)
	at org.betterx.betterend.registry.EndParticles.<clinit>(EndParticles.java:15)
	... 13 more


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Server thread
Stacktrace:
	at org.betterx.betterend.blocks.AncientEmeraldIceBlock.makeParticles(AncientEmeraldIceBlock.java:61)
	at org.betterx.betterend.blocks.AncientEmeraldIceBlock.method_9514(AncientEmeraldIceBlock.java:56)
	at net.minecraft.class_4970$class_4971.method_26199(class_4970.java:993)
	at net.minecraft.class_3218.method_18203(class_3218.java:506)
	at net.minecraft.class_3215.method_14161(class_3215.java:380)
	at net.minecraft.class_3215.method_12127(class_3215.java:324)
	at net.minecraft.class_3218.method_18765(class_3218.java:319)

-- Affected level --
Details:
	All players: 1 total; [class_3222['PurpleStar64'/304, l='ServerLevel[world]', x=903.40, y=139.08, z=943.45]]
	Chunk stats: 1751
	Level dimension: minecraft:the_end
	Derived: true
	Level spawn location: World: (0,74,32), Section: (at 0,10,0 in 0,4,2; chunk contains blocks 0,0,32 to 15,383,47), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,383,511)
	Level time: 10187 game time, 10187 day time
	Level name: world
	Level game mode: Game mode: survival (ID 0). Hardcore: false. Cheats: false
	Level weather: Rain time: 8375 (now: false), thunder time: 91116 (now: false)
	Known server brands: fabric
	Level was modded: true
	Level storage version: 0x04ABD - Anvil
Stacktrace:
	at net.minecraft.server.MinecraftServer.method_3813(MinecraftServer.java:901)
	at net.minecraft.class_3176.method_3813(class_3176.java:283)
	at net.minecraft.server.MinecraftServer.method_3748(MinecraftServer.java:828)
	at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:675)
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:266)
	at java.base/java.lang.Thread.run(Thread.java:1623)

-- System Details --
Details:
	Minecraft Version: 1.19.4
	Minecraft Version ID: 1.19.4
	Operating System: Windows 11 (amd64) version 10.0
	Java Version: 20.0.1, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode, sharing), Oracle Corporation
	Memory: 574152392 bytes (547 MiB) / 2042626048 bytes (1948 MiB) up to 6442450944 bytes (6144 MiB)
	CPUs: 32
	Processor Vendor: GenuineIntel
	Processor Name: 13th Gen Intel(R) Core(TM) i9-13980HX
	Identifier: Intel64 Family 6 Model 183 Stepping 1
	Microarchitecture: unknown
	Frequency (GHz): 2.42
	Number of physical packages: 1
	Number of physical CPUs: 24
	Number of logical CPUs: 32
	Graphics card #0 name: Parsec Virtual Display Adapter
	Graphics card #0 vendor: Parsec Cloud, Inc.
	Graphics card #0 VRAM (MB): 0.00
	Graphics card #0 deviceId: unknown
	Graphics card #0 versionInfo: DriverVersion=0.41.0.0
	Graphics card #1 name: NVIDIA GeForce RTX 4080 Laptop GPU
	Graphics card #1 vendor: NVIDIA (0x10de)
	Graphics card #1 VRAM (MB): 4095.00
	Graphics card #1 deviceId: 0x27e0
	Graphics card #1 versionInfo: DriverVersion=31.0.15.2866
	Graphics card #2 name: Intel(R) UHD Graphics
	Graphics card #2 vendor: Intel Corporation (0x8086)
	Graphics card #2 VRAM (MB): 1024.00
	Graphics card #2 deviceId: 0xa788
	Graphics card #2 versionInfo: DriverVersion=31.0.101.4091
	Memory slot #0 capacity (MB): 16384.00
	Memory slot #0 clockSpeed (GHz): 4.80
	Memory slot #0 type: Unknown
	Memory slot #1 capacity (MB): 16384.00
	Memory slot #1 clockSpeed (GHz): 4.80
	Memory slot #1 type: Unknown
	Virtual memory max (MB): 53890.70
	Virtual memory used (MB): 35484.26
	Swap memory total (MB): 21504.00
	Swap memory used (MB): 269.36
	JVM Flags: 1 total; -Xmx6G
	Fabric Mods: 
		appleskin: AppleSkin 2.4.3+mc1.19.4
		aqupdgrizzly: Grizzly Bear mob 1.19.4-1.5
		aqupdhat: Hat command 1.19.4-2.5.0
			server_translations_api: Server Translations API 2.0.0-beta.2+1.19.4-pre2
		audioplayer: AudioPlayer 1.19.4-1.5.2
		badpackets: Bad Packets 0.4.1
		bclib: BCLib 2.3.1
		bettercombat: Better Combat 1.6.2+1.19.4
		betterdeserttemples: YUNG's Better Desert Temples 1.19.4-Fabric-2.4.0
			org_reflections_reflections: reflections 0.10.2
		betterdungeons: YUNG's Better Dungeons 1.19.4-Fabric-3.4.0
		betterend: Better End 3.2.0
		betterfortresses: YUNG's Better Nether Fortresses 1.19.4-Fabric-1.2.0
		bettermineshafts: YUNG's Better Mineshafts 1.19.4-Fabric-3.4.0
		betternether: Better Nether 8.2.1
		betteroceanmonuments: YUNG's Better Ocean Monuments 1.19.4-Fabric-2.3.0
		betterstrongholds: YUNG's Better Strongholds 1.19.4-Fabric-3.4.0
		betterwitchhuts: YUNG's Better Witch Huts 1.19.4-Fabric-2.3.0
		botarium: Botarium 2.0.5
			team_reborn_energy: Energy 2.2.0
		carryon: Carry On 2.0.5.20
		cave-survey: Cave Survey 1.0.1
			ir_mehradn_common_: common 1.0.1
			polymer-core: Polymer (Core) 0.4.7+1.19.4
				polymer-networking: Polymer (Networking) 0.4.7+1.19.4
				polymer-registry-sync-manipulator: Polymer (Registry Sync Manipulator) 0.4.7+1.19.4
			polymer-resource-pack: Polymer (Resource Pack) 0.4.7+1.19.4
				polymer-common: Polymer (Common) 0.4.7+1.19.4
					packet_tweaker: Packet Tweaker 0.4.0+1.19.4
		cloth-config: Cloth Config v10 10.0.96
			cloth-basic-math: cloth-basic-math 0.6.1
		collective: Collective 6.54
		debugify: Debugify 1.19.4+2.2
			com_github_llamalad7_mixinextras: MixinExtras 0.2.0-beta.6
		emotecraft: Emotecraft 2.2.7-b.build.48
			player-animator: Player Animator 1.0.2+1.19.4
		enhancedgroups: Simple Voice Chat Enhanced Groups 1.19.4-1.3.0
		extended_drawers: Extended Drawers 1.3.9+mc.1.19.4
			graphlib: Graph Lib 0.5.0+1.19.4
			mconfig: Mconfig 1.2.0+mc.1.19-rc1
				de_poiu_apron_apron: apron 2.1.1
		fabric-api: Fabric API 0.79.0+1.19.4
			fabric-api-base: Fabric API Base 0.4.24+9ff28bcef4
			fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.6.25+49abcf7ef4
			fabric-biome-api-v1: Fabric Biome API (v1) 13.0.7+348a9c64f4
			fabric-block-api-v1: Fabric Block API (v1) 1.0.6+e022e5d1f4
			fabric-command-api-v1: Fabric Command API (v1) 1.2.27+f71b366ff4
			fabric-command-api-v2: Fabric Command API (v2) 2.2.6+e719b857f4
			fabric-commands-v0: Fabric Commands (v0) 0.2.44+df3654b3f4
			fabric-containers-v0: Fabric Containers (v0) 0.1.54+df3654b3f4
			fabric-content-registries-v0: Fabric Content Registries (v0) 3.5.9+ae0966baf4
			fabric-convention-tags-v1: Fabric Convention Tags 1.4.1+9a7c5daaf4
			fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.2.15+aeb40ebef4
			fabric-data-generation-api-v1: Fabric Data Generation API (v1) 11.4.0+6cebf059f4
			fabric-dimensions-v1: Fabric Dimensions API (v1) 2.1.45+7f87f8faf4
			fabric-entity-events-v1: Fabric Entity Events (v1) 1.5.13+e45f7c65f4
			fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.4.43+a1ccd7bff4
			fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.52+df3654b3f4
			fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.33+a1ccd7bff4
			fabric-item-api-v1: Fabric Item API (v1) 2.1.17+09a3510cf4
			fabric-item-group-api-v1: Fabric Item Group API (v1) 3.0.5+043f9acff4
			fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 2.2.15+5da15ca1f4
			fabric-loot-api-v2: Fabric Loot API (v2) 1.1.27+75e98211f4
			fabric-loot-tables-v1: Fabric Loot Tables (v1) 1.1.31+9e7660c6f4
			fabric-message-api-v1: Fabric Message API (v1) 5.1.1+1ee8be40f4
			fabric-mining-level-api-v1: Fabric Mining Level API (v1) 2.1.39+49abcf7ef4
			fabric-networking-api-v1: Fabric Networking API (v1) 1.3.1+a6f3ccfaf4
			fabric-networking-v0: Fabric Networking (v0) 0.3.41+df3654b3f4
			fabric-object-builder-api-v1: Fabric Object Builder API (v1) 7.0.3+63b515f4f4
			fabric-particles-v1: Fabric Particles (v1) 1.0.23+f1e4495bf4
			fabric-recipe-api-v1: Fabric Recipe API (v1) 1.0.8+a1ccd7bff4
			fabric-registry-sync-v0: Fabric Registry Sync (v0) 2.2.0+670e8ac6f4
			fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.3.28+afca2f3ef4
			fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 3.0.21+f1e4495bf4
			fabric-resource-conditions-api-v1: Fabric Resource Conditions API (v1) 2.3.1+e6c7d4eef4
			fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.11.2+1e1fb126f4
			fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.3.20+5da15ca1f4
			fabric-transfer-api-v1: Fabric Transfer API (v1) 3.1.1+da9bb835f4
			fabric-transitive-access-wideners-v1: Fabric Transitive Access Wideners (v1) 3.0.3+63b515f4f4
		fabric-language-kotlin: Fabric Language Kotlin 1.9.4+kotlin.1.8.21
			org_jetbrains_kotlin_kotlin-reflect: kotlin-reflect 1.8.21
			org_jetbrains_kotlin_kotlin-stdlib: kotlin-stdlib 1.8.21
			org_jetbrains_kotlin_kotlin-stdlib-jdk7: kotlin-stdlib-jdk7 1.8.21
			org_jetbrains_kotlin_kotlin-stdlib-jdk8: kotlin-stdlib-jdk8 1.8.21
			org_jetbrains_kotlinx_atomicfu-jvm: atomicfu-jvm 0.20.2
			org_jetbrains_kotlinx_kotlinx-coroutines-core-jvm: kotlinx-coroutines-core-jvm 1.6.4
			org_jetbrains_kotlinx_kotlinx-coroutines-jdk8: kotlinx-coroutines-jdk8 1.6.4
			org_jetbrains_kotlinx_kotlinx-datetime-jvm: kotlinx-datetime-jvm 0.4.0
			org_jetbrains_kotlinx_kotlinx-serialization-cbor-jvm: kotlinx-serialization-cbor-jvm 1.5.0
			org_jetbrains_kotlinx_kotlinx-serialization-core-jvm: kotlinx-serialization-core-jvm 1.5.0
			org_jetbrains_kotlinx_kotlinx-serialization-json-jvm: kotlinx-serialization-json-jvm 1.5.0
		fabricloader: Fabric Loader 0.14.19
		fabrictailor: Fabric Tailor 2.1.1
			config2brigadier: Config to Brigadier 1.2.3
			fabric-permissions-api-v0: fabric-permissions-api 0.2-SNAPSHOT
		fallingtree: FallingTree 3.12.2
		ferritecore: FerriteCore 5.2.0
		forgeconfigapiport: Forge Config API Port 6.0.2
			com_electronwill_night-config_core: core 3.6.5
			com_electronwill_night-config_toml: toml 3.6.5
		friendsandfoes: Friends&Foes 1.8.1
		ghastcow: The Ghast Cow mod 1.4.0
		goated: You've Goat to be kidding me 1.19.4-1.3.0
		grapplemod: Grappling Hook Mod 1.19.4-v1.5
		immersive_aircraft: Immersive Aircraft 0.4.2+1.19.4
		incendium: Incendium 5.2.1
		java: Java HotSpot(TM) 64-Bit Server VM 20
		kiwi: Kiwi Library 10.0.1
		krypton: Krypton 0.2.2
			com_velocitypowered_velocity-native: velocity-native 3.2.0-SNAPSHOT
		memoryleakfix: Memory Leak Fix 1.0.0
		midnightlib: MidnightLib 1.3.0
		mine_treasure_mr: Mine Treasure 1.2.7
		minecraft: Minecraft 1.19.4
		moonlight: Moonlight 1.19.4-2.3.4
		morediscs: More Music Discs 1.19.4-27-fabric
		moremobvariants: More Mob Variants 1.1.0
		mostructures: Mo' Structures 1.4.3+1.19.4
			omega-config: OmegaConfig 1.2.3-1.18.1
		nochatreports: No Chat Reports 1.19.4-v2.1.1
		nullscape: Nullscape 1.2.1
		phosphor: Phosphor 0.8.1
		random_mob_sizes: Random Mob Sizes 1.19.3-1.1.1
		resourcefulconfig: Resourcefulconfig 1.2.0
		sheets_mr: Bedsheets 1.3
		sit: Sit 1.19.4-23
		skeletonhorsespawn: Skeleton Horse Spawn 3.6
		sleep_tight: Sleep Tight 1.19.4-1.0.15
		smoothboot: Smooth Boot 1.19.4-1.7.0
		snowrealmagic: Snow! Real Magic! 8.0.1
		soulfired: Soul fire'd 3.1.0.0
		spawn_animations_mr: Spawn Animations 1.8.2
		subterrestrial: Subterrestrial 2.1.1
		terralith: Terralith 2.3.7b
		the_bumblezone: The Bumblezone - Fabric 6.7.14+1.19.4-fabric
			cardinal-components-base: Cardinal Components API (base) 5.1.0
			cardinal-components-entity: Cardinal Components API (entities) 5.1.0
			fake-player-api: Fake Player API 0.6.0
		torchhit: Torch hit! 6.0.1.2
		trinkets: Trinkets 3.6.0
		vcinteraction: Voice Chat Interaction 1.19.4-1.0.3
		voicechat: Simple Voice Chat 1.19.4-2.4.5
		winterly: Winterly 0.9.0
			cardinal-components-world: Cardinal Components API (worlds) 5.1.0
		wthit: wthit 7.2.3
		yungsapi: YUNG's API 1.19.4-Fabric-3.10.1
			org_javassist_javassist: javassist 3.29.2-GA
		yungsbridges: YUNG's Bridges 1.19.4-Fabric-3.3.0
		yungsextras: YUNG's Extras 1.19.4-Fabric-3.3.0
		zombiehorsespawn: Zombie Horse Spawn 4.6
	Server Running: true
	Player Count: 1 / 20; [class_3222['PurpleStar64'/304, l='ServerLevel[world]', x=903.40, y=139.08, z=943.45]]
	Data Packs: vanilla, Goated Generated Pack, fabric
	Enabled Feature Flags: minecraft:vanilla
	World Generation: Experimental
	Is Modded: Definitely; Server brand changed to 'fabric'
	Type: Dedicated Server (map_server.txt)
```

**Comment by DavidReiter** [comment]
+1 getting the same crash with version 1.19.4 on debian 11 server

**Comment by kowie-dev** [comment]
+1 here. 1.19.4 fabric

**Comment by Widechutz** [comment]
Hello
Ive experienced the same issue and have sat together with a friend to come up with a solution for it. We've made an edit to the mod to get rid of the particle that was causing the crashes. This edit is meant for serverside only as singleplayer doesnt have this bug.
https://github.com/Widechutz/BetterEnd/releases
Ignore all the rest of the stuff on the edited version (its broken, just use the releases)
The fix works for 1.19.3, for the other versions it should work but we haven't tested it out yet.
Hope it works and enjoy

**Comment by quiqueck** [comment]
Released with BetterEnd 3.1.3/3.2.1. If it does not fix the issue, please re-open.
